contributors/design-proposals/volume-provisioning.md

```diff
@@ -485,15 +533,55 @@ parameters:
    type: spinning
 ```
 
+### AWS provisioner and a PVC w
+
+This example shows how a PVC can require a volume from given zone. If there is
+a matching available PV, it will be bound to the PVC. If no such PV exists,
+a new one in the right zone will be provisioned by AWS provisioner.
+
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: myclaim
+  annotations:
+    "volume.beta.kubernetes.io/storage-class": "aws-fast"
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1500Mi
+  selector:
+    matchLabels:
+      "failure-domain.beta.kubernetes.io/zone": "us-east-1a"
+```
+
+When used with the StorageClass from the previous example, this PVC would never
+been provisioned as it requires zone "us-east-1a", while the class allows only
+"us-east-1b". It's up to the provisioner plugin to emit appropriate warning
+event on the PVC to let the user know why it can't be provisioned and how to fix
```

Comment:
events get lost.  is there any thought about having a `PersistentVolumeClaimCondition` in status that could report things like this now or in the future?